cpp/tests/test_abm_world.cpp

```diff
@@ -165,7 +170,7 @@ TEST(TestWorld, evolveStateTransition)
     EXPECT_EQ(p3.get_infection_state(t + dt), mio::abm::InfectionState::InfectedSymptoms);
 }
 
-TEST(TestWorld, evolveMigration)
+TEST(TestWorld, evolveMigration) // TODO: this is not a unit test, this is a unit of a test!
```

Comment:
Yes, it basically tests certain aspects of the evolve function, which is basically all other big functions together. Do you have ideas how to make a simpler unit test for this?

Reply:
I think it would be a lot better to split the test. The idea for a unit test is to have a test case (i.e. both the "question" as well as the desired result) for a small unit of code. The evolve function mostly just calls other functions, so there is little to test there, instead, we should have a test `evolveBeginStepAspect1`, `evolveInteractAspect2`, and so on. Or skip the evolve part when we can access the "aspect" directly.

What is really important IMO, is that the test case has to be clear. The only thing I could tell you about "evolveMigration" is *that* it covers evolve. It is way to long for me to understand *how* without investing a lot of time. Also, but that is sadly true for most of our tests, it is unclear whether the test has an expected answer, or just a copy of the answer the code gave at the time of implementation (This is what I call a "consistency test". They can be useful, but you usually do not want one).

Reply:
I opened #1067 so we can address this later. I'll remove the TODO, as it is out of scope for this issue.